toot-relay.go

```diff
@@ -21,12 +21,54 @@ import (
 	"golang.org/x/net/http2"
 )
 
+type Message struct {
+	isProduction bool
+	notification *apns2.Notification
+}
+
 var (
 	developmentClient *apns2.Client
 	productionClient  *apns2.Client
 	topic             string
+	messageChan       chan *Message
+)
+
+const (
+	maxWorkers     = 4
+	maxQueueLength = 1024
 )
 
+func worker(workerId int) {
+	log.Printf("starting worker %d", workerId)
+	defer log.Printf("stopping worker %d", workerId)
+
+	var client *apns2.Client
+
+	for msg := range messageChan {
+		if msg.isProduction {
+			client = productionClient
+		} else {
+			client = developmentClient
+		}
+
+		res, err := client.Push(msg.notification)
+
+		if err != nil {
+			log.Println("Push error:", err)
+			return
```

Comment:
I think this has to be a `continue`, otherwise you're exiting the worker?

Reply:
Ah, you're totally right. That's a holdover for when I had the whole push in a goroutine. I can update it in by fork if you'd like, or you can just swap it.